cli/cli/helpers/portal_manager/portal_manager.go

```
@@ -33,15 +46,139 @@ func (portalManager *PortalManager) GetClient() portal_generated_api.KurtosisPor
+}
+
+// StopExisting stops the existing Portal process, if any, and removes the PID file
+// It returns an error if something went wrong
+func (portalManager *PortalManager) StopExisting(_ context.Context) error {
+	portalPidFilePath, err := host_machine_directories.GetPortalPidFilePath()
+	if err != nil {
+		return stacktrace.Propagate(err, "Unable to get file path to PID file")
+	}
+
+	pid, err := getPidFromPidFile()
+	if err != nil {
+		return stacktrace.Propagate(err, "Error getting current Portal PID")
+	}
+	if pid == 0 {
+		// Not running, nothing to do
+		if portalManager.IsReachable() {
+			logrus.Warnf("Kurtosis Portal process not registered in Kurtosis, but it seems a Portal can be " +
+				"reached locally on its ports. This is unexpected and Kurtosis cannot stop it. Was the Portal started " +
+				"with something else then Kurtosis CLI? If that's the case, please kill the current Portal process and " +
+				"start it using Kurtosis CLI")
+			return nil
+		}
+		return nil
+	}
+
+	process, err := getRunningProcessFromPid(pid)
+	if err != nil {
+		return stacktrace.Propagate(err, "Error getting Portal process from its PID %d", pid)
+	}
+	if process == nil {
+		if portalManager.IsReachable() {
+			logrus.Warnf("Kurtosis Portal process not registered in Kurtosis, but it seems a Portal can be " +
+				"reached locally on its ports. This is unexpected and Kurtosis cannot stop it. Was the Portal started " +
+				"with something else then Kurtosis CLI? If that's the case, please kill the current Portal process and " +
+				"start it using Kurtosis CLI")
+			return nil
+		}
+	}
+
+	if err = process.Signal(syscall.SIGINT); err != nil {
+		logrus.Warnf("Error stopping currently running portal on PID: '%d'. It might already be stopped. "+
+			"PID file will be removed. Error was: %s", pid, err.Error())
+	}
+
+	if err = os.Remove(portalPidFilePath); err != nil {
+		return stacktrace.Propagate(err, "Portal was successfully stopped but Kurtosis couldn't remove the "+
```

perhaps log an error instead of failing? but that doesn't return a non 0 exit code

---

Yup that seems right to not fail loudly here 👍